src/api/stats.ts

```diff
@@ -134,6 +190,43 @@ const getStatsForBilling = (tenants: string[], startDate: string) => {
         .order('ts', { ascending: false });
 };
 
+const getStatsForDetails = (
+    catalogName: string,
+    entityType: Entity,
+    grain: string,
+    duration?: Duration
+) => {
+    const today = new Date();
+    const past = duration ? sub(today, duration) : today;
+
+    const gt = formatToGMT(startOfHour(past), true);
+    const lte = formatToGMT(startOfHour(today), true);
+
+    let query: string;
+    switch (entityType) {
+        case 'capture':
+            query = CAPTURE_QUERY;
+            break;
+        case 'materialization':
+            query = MATERIALIZATION_QUERY;
+            break;
+        case 'collection':
+            query = COLLECTION_QUERY;
+            break;
+        default:
+            query = DEFAULT_QUERY;
+    }
+
+    return supabaseClient
+        .from<CatalogStats_Billing>(TABLES.CATALOG_STATS)
+        .select(query)
+        .eq('catalog_name', catalogName)
+        .eq('grain', grain)
+        .gt('ts', gt)
```

Comment:
Nope - for the reason you stated. Since we're fetching specific hours we need to actually make sure they are greater than so we don't fetch 1 extra element every query.